include/eve/algo/container/detail/soa_storage2.hpp

```diff
@@ -0,0 +1,172 @@
+    soa_storage(soa_storage const& src) : storage_(src.storage), size_(src.size())
+    {
+      reindex(src.size());
+    }
+
+    soa_storage& operator=(soa_storage const& src)
+    {
+      storage_ = src.storage;
+      size_ = src.size();
+      reindex(src.size());
+      return *this;
+    }
+
+    soa_storage(soa_storage&&)            =default;
+    soa_storage& operator=(soa_storage&&) =default;
+
+    ~soa_storage()
+    {
+      using alloc_t = std::allocator_traits<Allocator>;
+      kumi::for_each( [&]<typename T>(T*& s)
+                      {
+                        for(std::size_t i=0;i<size_;++i) alloc_t::destroy(*this, s+i);
+                      }
+                    , indexes_
+                    );
+    }
+
+    EVE_FORCEINLINE auto data_aligned()
+    {
+      auto ptrs = kumi::map([](auto* m) { return as_aligned_pointer(m); }, indexes_);
+      return kumi::apply(eve::algo::views::zip, ptrs);
+    }
+
+    EVE_FORCEINLINE auto data_aligned() const
+    {
+      auto ptrs = kumi::map([](auto const* m) { return as_aligned_pointer(m); }, indexes_);
+      return kumi::apply(eve::algo::views::zip, ptrs);
+    }
+
+    EVE_FORCEINLINE auto data()       { return eve::unalign(data_aligned()); }
+    EVE_FORCEINLINE auto data() const { return eve::unalign(data_aligned()); }
+
+    void swap(soa_storage& other) noexcept
+    {
+      storage_.swap(other.storage_);
+      std::swap(indexes_,other.indexes_);
+      std::swap(size_,other.size_);
+    }
+
+    void build(std::size_t n, Type v = {})
+    {
+      using alloc_t = std::allocator_traits<Allocator>;
+      auto sub_span = eve::align(&storage_[0], eve::over{64});
+
+      kumi::for_each( [&]<typename T>(T*& s, auto m)
+                      {
+                        s = reinterpret_cast<T*>(sub_span);
+                        for(std::size_t i=0;i<n;++i) alloc_t::construct(*this, s+i,m);
+                        sub_span += realign<T>(n);
+                      }
+                    , indexes_, v
+                    );
+
+      capacity_ = capacity_impl(n);
+    }
+
+    void reindex(std::size_t n)
+    {
+      auto sub_span = eve::align(&storage_[0], eve::over{64});
+
+      kumi::for_each( [&]<typename T>(T*& s)
+                      {
+                        s = reinterpret_cast<T*>(sub_span);
+                        sub_span += realign<T>(n);
+                      }
+                    , indexes_
+                    );
+      capacity_ = capacity_impl(n);
+    }
+
+    //==============================================================================================
+    // Static helpers
+    //==============================================================================================
+    template<typename Content>
+    static void piecewise_copy(Content& dst, Content const& src, std::size_t n)
+    {
+      kumi::for_each( [&]<typename T>(T* d, T const* s) { std::memcpy(d,s,sizeof(T)*n); }
+                    , dst, src
+                    );
+    }
```

Comment:
I don't know what you are doing, but this looks like `eve::algo::copy` only badly implemented.

Reply:
Indeed.

Reply:
Fixed